src/main/java/ca/uhn/fhir/jpa/starter/service/HelperService.java

```diff
@@ -1294,7 +1327,7 @@ public ThreadPoolTaskExecutor getAsyncExecutor() {
 	public ResponseEntity<?> getAsyncData(Map<String, String> categoryWithHashCodes, Boolean isAnonymizationEnabled) throws SQLException, IOException {
 		List<DataResultJava> dataResult = new ArrayList<>();
 		for (Map.Entry<String, String> item : categoryWithHashCodes.entrySet()) {
-			String dateRangeValue = getFormattedUUID(item.getValue());
+			String dateRangeValue = getFormattedUuid(item.getValue());
```

Comment:
If this regular expression is for getting the start and end date , you can directly get it from client in the request payload

Reply:
This regular expression is to get the date range with category id and we are getting it from the categoryWithHashCodes which is created using the request payload.